### DISCUSSION TITLE
Foreground Service won't keep a websocket connected after 5 minutes.

### DISCUSSION THREAD

**Discussion by eledumen** [capawesome-team/capacitor-plugins#147] [Q&A] [answered]
Hello,

I'm currently utilizing the SignalR JavaScript library (Websockets) to enable server-push data to my Ionic app. To ensure continuous data reception, even when users switch to other apps, I've incorporated the Foreground Service plugin.

However, I'm facing an issue where the connection is lost approximately 5 minutes after the app is minimized. I'm wondering if the Foreground Service plugin is supposed to keep the app WebView active despite system optimizations. Am I overlooking something?

Your insights and suggestions would be greatly appreciated. Thank you!

**Comment by robingenz** [comment, marked as answer]
> I'm wondering if the Foreground Service plugin is supposed to keep the app WebView active despite system optimizations. Am I overlooking something?

No, only the Foregrund service is active in the background, not the WebView. You need a native implementation for the websocket connection.
